crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -195,34 +195,37 @@ fn merge_stream_and_receiver(
 
         loop {
             select! {
-                biased; // Prefer manual updates for easier test code
+                // We want to give priority on updates from `raw_stream` as it will necessarily trigger a “refresh” of the rooms.
+                biased;
```

Comment:
Any way to add a regression test for the issue fixed by 98aae99674def3e38505f71c464846de850237ba?

Reply:
In a next PR yes. I would like to clearly tests the room info update in a dedicated test suite.

Reply:
Not sure how it's fixing anything though; there's no await in the select's body, so no place for tokio to interrupt a branch or the other. I really would like to understand it better, and a regression test would be an irrefutable proof.